### DISCUSSION TITLE
Windows 11 Build Error

### DISCUSSION THREAD

**Discussion by A6i8** [AcademySoftwareFoundation/OpenRV#396] [Build Issues] [answered]
I followed the Windows build guide carefully but i am getting error with **RV_DEPS_FFMPEG** dependencie.

But RV_DEPS_DAV1D build successfully.

`
Done Building Project "C:\OpenRV\_build\cmake\dependencies\RV_DEPS_DAV1D.vcxproj"
`

**RV_DEPS_FFMPEG Error**

`
error : dav1d >= 0.5.0 not found using pkg-config [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_FFMPEG.vcxproj]
         
         If you think configure made a mistake, make sure you are using the latest
         version from Git.  If the latest version fails, report the problem to the
         ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
         Include the log file "ffbuild/config.log" produced by configure as this will help
         solve the problem.
    52>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'C:\OpenRV\_build\CMakeFiles\cfe1b385b767f99aba28de75c1583622\RV_DEPS_FFMPEG-configure.rule;C:\OpenRV\_build\CMakeFiles\cfe1b385b767f99aba28de75c1583622\RV_DEPS_FFMPEG-build.rule;C:\OpenRV\_build\CMakeFiles\cfe1b385b767f99aba28de75c1583622\RV_DEPS_FFMPEG-install.rule;C:\OpenRV\_build\CMakeFiles\ddfe0fbca828d26d63401d1d79fd8a3b\RV_DEPS_FFMPEG-complete.rule;C:\OpenRV\_build\CMakeFiles\dc0b82080efe119cccb8b58ef9885864\RV_DEPS_FFMPEG.rule' exited with code 1. [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_FFMPEG.vcxproj]
    52>Done Building Project "C:\OpenRV\_build\cmake\dependencies\RV_DEPS_FFMPEG.vcxproj" (default targets) -- FAILED.
`

**rvcfg**

-- Using atomic_ops:   7.7.0
-- Using Boost:        1.80.0
-- Using Dav1d:        1.0.0
-- Using FFMPEG:       n4.4.3
-- Using atomic_ops:   7.7.0
-- Using Imath:        3.1.5
-- Using JpegTurbo:    2.1.4
-- Using OpenEXR:      3.1.7
-- Using OpenColorIO   2.2.1
-- Using OpenImageIO   2.4.6.0
-- Using OpenJpeg:     2.5.0
-- Using OpenSSL:      1.1.1u
-- Using PNG:          1.6.39
-- Using Python3:      3.10.12
-- Using PySide2:      5.15.10
-- Using Qt5:          5.15.2
-- Using Raw:          0.21.1
-- Using Tiff:         4.6.0
-- Using WebP:         1.2.1
-- Using Yaml-CPP:     0.7.0 (Using OCIO's own copy)
-- Using zlib:         1.2.13
-- Using spdlog:       1.11.0

@geffrak

**Comment by cedrik-fuoco-adsk** [comment]
Hello @A6i8!

There's a in-progress PR that will fix that issue.
You can look at the PR here : https://github.com/AcademySoftwareFoundation/OpenRV/pull/385

**Comment by cedrik-fuoco-adsk** [comment, marked as answer]
The PR has been merged. Please try again with the latest main branch and let me know if it is resolved. @A6i8

**Comment by yeoldebrian** [comment]
Hi @cedrik-fuoco-adsk I have the latest code and am getting a similar error:

```powershell
         C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'E:\OpenRV\_build\CMakeFiles\ba9056a2e12bbb280c19368b23728de1\RV_DEPS_PYTHON3-requ
       irements-flag.rule;E:\OpenRV\_build\CMakeFiles\ba9056a2e12bbb280c19368b23728de1\RV_DEPS_PYSIDE2-build-flag.rule;E:\OpenRV\_build\CMakeFiles\ed214e5487734762d00e67784b1bbf86\python310.dll.rule;E:\OpenRV\_build\CMakeFiles\b2670343e3e6d1
       3858c550f7afcec460\RV_DEPS_PYTHON3-stage-target.rule;E:\OpenRV\cmake\dependencies\CMakeLists.txt' exited with code 9009. [E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
```

I had six errors prior to this one and used this answer to get down to just the 1 error. tl/dr I edited my python3.cmake to point to my windows python 3.10 install instead of using the msys python:
https://github.com/AcademySoftwareFoundation/OpenRV/discussions/362

p.s. these are the 6 errors I get when I use the msys python:
```powershell
       (CustomBuild target) ->
         CUSTOMBUILD : error : Invalid license key ()! [E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         CUSTOMBUILD : error : Licensee info missing! [E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         CUSTOMBUILD : error : Licensed product not known ()! [E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         CUSTOMBUILD : Project error : License check failed! Giving up ... [E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         CUSTOMBUILD : error : Option --jobs can only be used with 'jom' on Windows. [E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
         C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'E:\OpenRV\_build\CMakeFiles\ba9056a2e12bbb280c19368b23728de1\RV_DEPS_PYSIDE2-buil
       d-flag.rule;E:\OpenRV\_build\CMakeFiles\ed214e5487734762d00e67784b1bbf86\python310.dll.rule;E:\OpenRV\_build\CMakeFiles\b2670343e3e6d13858c550f7afcec460\RV_DEPS_PYTHON3-stage-target.rule' exited with code 1. [E:\OpenRV\_build\cmake\de
       pendencies\RV_DEPS_PYTHON3-stage-target.vcxproj]

```

**Comment by cedrik-fuoco-adsk** [comment]
@yeoldebrian Could you provide the full log? Sometime the actual errors are buried in the middle of the logs. 

Also, instead of modifying the python3.cmake, you could add the correct python location to the beginning of the PATH env. variable (that's what I am doing).

And for those 6 errors, those errors normally come up when it is picking up an incompatible license (or missing) for Qt when building pyside2.

**Comment by yeoldebrian** [comment]
@cedrik-fuoco-adsk Thanks so much for responding. I thought the license error might have something to do with QT. I have a .qt-license wiht a key in it but maybe it's expired? Should I be using the qt for open source or can I just acquire another trial license?

Here's everything that happened after I typed "rvbootstrap" if that's what you're asking for?
[rvbootstrapLog.txt](https://github.com/AcademySoftwareFoundation/OpenRV/files/14409639/rvbootstrapLog.txt)

**Comment by cedrik-fuoco-adsk** [comment]
> Hi @cedrik-fuoco-adsk I have the latest code and am getting a similar error:
> 
> ```powershell
>          C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'E:\OpenRV\_build\CMakeFiles\ba9056a2e12bbb280c19368b23728de1\RV_DEPS_PYTHON3-requ
>        irements-flag.rule;E:\OpenRV\_build\CMakeFiles\ba9056a2e12bbb280c19368b23728de1\RV_DEPS_PYSIDE2-build-flag.rule;E:\OpenRV\_build\CMakeFiles\ed214e5487734762d00e67784b1bbf86\python310.dll.rule;E:\OpenRV\_build\CMakeFiles\b2670343e3e6d1
>        3858c550f7afcec460\RV_DEPS_PYTHON3-stage-target.rule;E:\OpenRV\cmake\dependencies\CMakeLists.txt' exited with code 9009. [E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
> ```

Is that the log that has the error above (and not using msys python)? I don't see it in the log.

If you installed Qt with the qt online installer, you can try to just open it up to the download page. When the installer starts, I think it will refresh the license if possible.

**Reply by yeoldebrian** [reply]
The error log I uploaded was using the msys python, I thought it was getting further and that's why it had the license errors? 

I changed out my python path in "C:\msys64\home\brian\.bashrc" and now when I rvbootstrap it says "Build succeeded" with 56 warnings but when I run rvtest it fails everything, Here's the log of everything with the test at the end.
[rvbootstrapLog.txt](https://github.com/AcademySoftwareFoundation/OpenRV/files/14424656/rvbootstrapLog.txt)
And I don't know where to find the executable or what to do next.

I'm trying to install open source qt but the download is very very slow, probably because it's an archive. I can't figure out how to refresh my license with the current install. Will open source QT work too?
Thanks again for you help!

**Comment by cedrik-fuoco-adsk** [comment]
You can find the OpenRV executable in `stage\app\bin` under your build folder, which should be `E:\OpenRV\_build\stage\app\bin\rv.exe` based on your logs.

Yes, Open source Qt should work with OpenRV.

**Comment by yeoldebrian** [comment]
Oh wow, yes it works, thank you! What files do I need just to run rv.exe, everything in the app folder?

**Reply by cedrik-fuoco-adsk** [reply]
Yes, it is everything under the app folder. You can also use the `rvinst` alias from rvcmd.sh to install (copy) it to `_install` directory under your OpenRV folder. It will copy everything needed.

**Comment by yeoldebrian** [comment]
Thanks again for all your help! I know this is the wrong place to ask about this but I'd love some guidance on getting Apple prores to work. I tried commenting out the disallowedCodecsArray in init.cpp but that didn't have any impact:
```cpp
// static const char* disallowedCodecsArray[] = {
//     "ac3",
//     "hevc",
//     "mpeg2video",
//     "prores",
//     "prores_aw",
//     "prores_ks",
//     "prores_lgpl",
//     "svq1",
//     "svq3",
//     0 };

static const char* disallowedCodecsArray[] = {0};
```

**Reply by bernie-laberge** [reply]
You also have to comment out this line:
https://github.com/AcademySoftwareFoundation/OpenRV/blob/bd7058b42cae8d22f73b6b1f40b9a08fb66deba1/cmake/dependencies/ffmpeg.cmake#L207

**Reply by bernie-laberge** [reply]
Please let us know if you still have a Windows build issue or not. Thanks!

**Comment by yeoldebrian** [comment]
That did it, thank you so much @bernie-laberge!
